Architecture_Engine/Modify/SetInternalElements2D.cs

```diff
@@ -0,0 +1,52 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Dimensional;
+using BH.oM.Geometry;
+using BH.oM.Architecture.Elements;
+using System.Collections.Generic;
+using System.Linq;
+
+using BH.oM.Reflection.Attributes;
+using System.ComponentModel;
+
+namespace BH.Engine.Architecture
+{
+    public static partial class Modify
+    {
+        /***************************************************/
+        /****               Public Methods              ****/
+        /***************************************************/
+
+        [Description("Assign a new collection of internal 2D elements to an Architecture Room")]
+        [Input("room", "An Architecture Room to update")]
+        [Input("internalElements2D", "A collection of internal 2D elements to assign to the panel")]
+        [Output("room", "The updated Architecture Room")]
+        public static Room SetInternalElements2D(this Room room, List<IElement2D> internalElements2D)
+        {
+            Room rr = room.GetShallowClone() as Room;
+            rr.Perimeter = internalElements2D.Cast<ICurve>().ToList()[0];
+            return rr;
+        }
```

Comment:
This looks like it's replacing the the rooms perimeter with a new one, which is not the point of the method. 

I'm unsure of how the location point ties into this but as a Room is defined by a perimeter it can never have any internal elements and should simply return the room here

Reply:
Should clarify that attempting to set internal elements to a IElement2d which does not have any should return a error.
i.e. if `internalElements2D` Count isn't zero, there should be an error